src/hip/hip_build_utils.cpp

```diff
@@ -174,20 +174,20 @@ static fs::path HipBuildImpl(boost::optional<TmpDir>& tmp_dir,
 #endif
 
     // hip version
-    params +=
-        std::string(" -DHIP_PACKAGE_VERSION_FLAT=") + std::to_string(HIP_PACKAGE_VERSION_FLAT);
-
-    params += " ";
+    params += " -DHIP_PACKAGE_VERSION_FLAT=" MIOPEN_STRINGIZE(HIP_PACKAGE_VERSION_FLAT) " ";
     auto bin_file = make_object_file_name(tmp_dir.get() / filename);
 
     // compile
     {
-        const std::string redirector = testing_mode ? " 1>/dev/null 2>&1" : "";
-        const std::string cmd        = env + std::string(" ") + MIOPEN_HIP_COMPILER;
-        const std::string args       = params + filename + " -o " + bin_file + redirector;
-        tmp_dir->Execute(cmd, args);
+        std::string args = params + filename + " -o " + bin_file;
+#ifndef _WIN32
+        // Windows uses WIN32 API to execute a subprocess. None command shell is spawned.
+        if(testing_mode)
+            args += " 1>/dev/null 2>&1";
+#endif
+        tmp_dir->Execute(MIOPEN_HIP_COMPILER, args, env);
```

Comment:
[Notice] @apwojcik AFAICS `env` remains empty. Seems like this is a rudiment of the past. So the support for setting the enviroment in `TmpDir::Execute()` is not really needed. I guess you may wish to remove it to simplify the code. Of course, this is up to you.